hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/FileGroupReaderBasedMergeHandle.java

```diff
@@ -90,7 +147,15 @@ public FileGroupReaderBasedMergeHandle(HoodieWriteConfig config, String instantT
     this.maxInstantTime = maxInstantTime;
     this.keyToNewRecords = Collections.emptyMap();
     this.readerContext = readerContext;
-    this.operation = operation;
+    this.operation = Option.of(operation);
+    initRecordTypeAndCdcLogger(enginRecordType);
+    init(operation, this.partitionPath);
+    this.props = TypedProperties.copy(config.getProps());
+    this.isCompaction = true;
+    initRecordIndexCallback();
```

Comment:
do we even need to track RLI for compactions?

Reply:
Removed this, it was causing some issues with existing tests so it was good to see we have some good coverage here :)